dipy/reconst/dki.py

```diff
@@ -0,0 +1,743 @@
+            All parameters estimated from the diffusion kurtosis model.
+            Parameters are ordered as follow:
+                1) Three diffusion tensor's eingenvalues
+                2) Three lines of the eigenvector matrix each containing the
+                   first, second and third coordinates of the eigenvector
+                3) Fifteen elements of the kurtosis tensor
+
+        gtab : a GradientTable class instance
+            The gradient table for this prediction
+
+        S0 : float or ndarray (optional)
+            The non diffusion-weighted signal in every voxel, or across all
+            voxels. Default: 1
+
+        Notes
+        -----
+        The predicted signal is given by:
+
+        .. math::
+
+            S(n,b)=S_{0}e^{-bD(n)+\frac{1}{6}b^{2}D(n)^{2}K(n)}
+
+        $\mathbf{D(n)}$ and $\mathbf{K(n)}$ can be computed from the DT and KT
+        using the following equations:
+
+        .. math::
+
+            D(n)=\sum_{i=1}^{3}\sum_{j=1}^{3}n_{i}n_{j}D_{ij}
+
+        and
+
+        .. math::
+
+            K(n)=\frac{MD^{2}}{D(n)^{2}}\sum_{i=1}^{3}\sum_{j=1}^{3}
+            \sum_{k=1}^{3}\sum_{l=1}^{3}n_{i}n_{j}n_{k}n_{l}W_{ijkl}
+
+        where $D_{ij}$ and $W_{ijkl}$ are the elements of the second-order DT
+        and the fourth-order KT tensors, respectively, and $MD$ is the mean
+        diffusivity.
+        """
+        return dki_prediction(self.model_params, self.gtab, S0)
+
+
+def ols_fit_dki(design_matrix, data, min_signal=1):
+    r""" Computes ordinary least squares (OLS) fit to calculate the diffusion
+    tensor and kurtosis tensor using a linear regression diffusion kurtosis
+    model [1]_.
+
+    Parameters
+    ----------
+    design_matrix : array (g, 22)
+        Design matrix holding the covariants used to solve for the regression
+        coefficients.
+    data : array (N, g)
+        Data or response variables holding the data. Note that the last
+        dimension should contain the data. It makes no copies of data.
+    min_signal : default = 1
+        All values below min_signal are repalced with min_signal. This is done
+        in order to avoid taking log(0) durring the tensor fitting.
+
+    Returns
+    -------
+    dki_params : array (N, 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+
+    See Also
+    --------
+    wls_fit_dki
+
+    References
+    ----------
+       [1] Tabesh, A., Jensen, J.H., Ardekani, B.A., Helpern, J.A., 2011.
+           Estimation of tensors and tensor-derived measures in diffusional
+           kurtosis imaging. Magn Reson Med. 65(3), 823-836
+    """
+
+    tol = 1e-6
```

Comment:
Could we make the tolerance an argument, with this default value, rather than have it be a hard-coded constant? I understand this is a tolerance that defines the minimal diffusivity allowed in the solution. Is that correct? Maybe give this variable a more informative name?